routes/golfer_search.go

```diff
@@ -0,0 +1,44 @@
+package routes
+
+import (
+	"net/http"
+
+	"github.com/code-golf/code-golf/config"
+	"github.com/code-golf/code-golf/session"
+)
+
+// GET /golfer/search
+func golferSearchGET(w http.ResponseWriter, r *http.Request) {
+	data := struct {
+		Langs     map[string]string
+		Holes     map[string]string
+		Solutions []struct {
+			Code    string `json:"code"`
+			Hole    string `json:"hole"`
+			Lang    string `json:"lang"`
+			Scoring string `json:"scoring"`
+		}
+	}{
+		Langs: make(map[string]string),
+		Holes: make(map[string]string),
+	}
+
+	for k, v := range config.AllLangByID {
+		data.Langs[k] = v.Name
+	}
+	for k, v := range config.AllHoleByID {
```

Comment:
AllHoleByID includes exp, which won't have solutions in the db, just use HoleByID